### PR TITLE
Add configuration to require POST only authentication requsts

### PR DESCRIPTION
**Pull request by sultaniman**
## Proposed changes

This PR introduces a new configuration and authentication behaviour when it is active and will require POST request during authentication

* Configuration key is `SOCIAL_AUTH_REQUIRE_POST: bool`,
* Error response if request method is not `POST`.

## Types of changes

Please check the type of change your PR introduces:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Related issues and discussions

* https://github.com/python-social-auth/social-app-django/issues/486
* https://github.com/python-social-auth/social-app-django/issues/55

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: social_django/utils.py [hunk 1]
```diff
@@ -48,3 +50,15 @@ def wrapper(request, backend, *args, **kwargs):
...
+def maybe_require_post(func):
+    @wraps(func)
+    def wrapper(request, backend, *args, **kwargs):
+        require_post = getattr(settings, REQUIRE_POST, False)
+        if require_post and request.method != "POST":
+            return HttpResponseNotAllowed(["POST"])
+
+        return func(request, backend, *args, **kwargs)
+
+    return wrapper
```
Comment (nijel) [inline social_django/utils.py:64]: Maybe this could reuse require_POST from Django instead of reimplementing it?

Comment (sultaniman) [reply social_django/utils.py:64]: @nijel did you mean something like this?
```py
if require_post:
    return require_POST(func)(request, backend, *args, **kwargs)
```

Comment (nijel) [reply social_django/utils.py:64]: Yes, something like this.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by sultaniman** [comment]
@nijel I updated docs and here is the [PR](https://github.com/python-social-auth/social-docs/pull/174)

**Pull request merged**